src/service.ts

```diff
@@ -186,46 +225,55 @@ const defaultCLIArguments: CLIArguments = {
   ignorePath: ".prettierignore",
 };
 
-function parseCLIArguments(args: string[]): [CLIArguments, string] {
+function parseCLIArguments(args: string[]): [CLIArguments, string, CliOptions] {
   const parsedArguments: CLIArguments = { ...defaultCLIArguments };
   let fileName: string | null = null;
 
+  const optionArgs: string[] = [];
+
   const argsIterator = args[Symbol.iterator]();
   for (const arg of argsIterator) {
-    switch (arg) {
-      case "--no-color":
-        parsedArguments.noColor = true;
-        break;
-
-      case "--ignore-path": {
-        const nextArg = argsIterator.next();
-        if (nextArg.done) {
-          throw new Error("--ignore-path option expects a file path");
+    if (arg.startsWith("-")) {
+      switch (arg) {
+        case "--no-color":
+          parsedArguments.noColor = true;
+          break;
+
+        case "--ignore-path": {
+          const nextArg = argsIterator.next();
+          if (nextArg.done) {
+            throw new Error("--ignore-path option expects a file path");
+          }
+
+          parsedArguments.ignorePath = nextArg.value;
+          break;
         }
-
-        parsedArguments.ignorePath = nextArg.value;
-        break;
-      }
-
-      default:
-        // NOTE: positional arguments are assumed to be file paths
-        if (fileName) {
-          throw new Error("Only a single file path is supported");
+        default: {
+          optionArgs.push(arg);
```

Comment:
This won't work if the user does something like `--print-width 80` right? Like it'll interpret 80 as a filename?

Reply:
Yeah, supporting that would require a bit more code. Do you want me to do it in this PR?

Reply:
To be honest, I think we can do that in a follow-up, or later. To solve for that we would need to keep track of all flags that take a parameter, right? Or requre the user to use `=` (i.e. `--print-width=80` instead of `--print-width 80`), I don't love that though.

Reply:
> To solve for that we would need to keep track of all flags that take a parameter, right?

Not really. We just check if the next arg starts with `-`, if yes treat the current arg as boolean flag, otherwise consider the next arg as the value for current arg. That should be good enough.

If the user doesn't provide value for a flag that expects it, e.g. `--print-width --no-semi`, then we'll end up passing `{ printWidth: true, semi: false }` to prettier, and prettier will throw error (which is expected).

So it'll be the user's responsibility to provide correct flags.

Reply:
> Not really. We just check if the next arg starts with -, if yes treat the current arg as boolean flag, otherwise consider the next arg as the value for current arg. That should be good enough.

We can't do that. Think of `prettierd --no-semi file.ts`

Reply:
Hmmm... Yeah...

Can we force the filename to be the first positional arg? Or it would've been easier if filepath was taken as the value of `--stdin-filepath`.

Keeping a list of all the flags that take value is kinda impossible, unless prettier exposes an api for it. Because any plugin can add new options and prettier accepts those options as cli flags. For example, if you're using this plugin: https://github.com/un-ts/prettier/tree/master/packages/sql, you can pass `--keyword-case` flag with the supported values (`'preserve' | 'upper' | 'lower'`).

Reply:
Yeah right now I'm debating whether we push the breaking change and go for consistency with prettier (i.e. use `--stdin-filename`) or force the last argument to be the filename and keep things backwards compatible (though we'd require the flags to come first).

Alternatively we can do it in two steps: introduce `--stdin-filename` and then push users towards the flag with a warning or something like that? Then eventually remove support for the positional argument (problem is that I don't know if the warning will even be visible as folks usually use prettied within editors).